src/wallet/walletutil.cpp

```diff
@@ -60,10 +61,16 @@ std::vector<fs::path> ListWalletDir()
     const size_t offset = wallet_dir.string().size() + 1;
     std::vector<fs::path> paths;
     boost::system::error_code ec;
+    boost::system::error_code eci;
 
-    for (auto it = fs::recursive_directory_iterator(wallet_dir, ec); it != fs::recursive_directory_iterator(); it.increment(ec)) {
+    for (auto it = fs::recursive_directory_iterator(wallet_dir, ec); it != fs::recursive_directory_iterator(); it.increment(eci)) {
         if (ec) {
-            LogPrintf("%s: %s %s\n", __func__, ec.message(), it->path().string());
+            LogPrintf("%s: iterator: %s %s\n", __func__, ec.message(), it->path().string());
+            continue;
+        }
+        if (eci) {
+            LogPrintf("%s: increment: %s %s\n", __func__, eci.message(), it->path().string());
+            it.disable_recursion_pending();
```

Comment:
`disable_recursion_pending` is a new name introduced in Boost 1.72. We support much older. Renaming this to `no_push` (the old name) should just work.